File: 07-07-2024_Water_Bottles.java
class Solution {
    public int numWaterBottles(int numBottles, int exc) {
        int empty = numBottles, ans = numBottles;
        while(empty >= exc){
            int filled  = empty / exc;
            int rem = empty % exc;
            ans += filled;
            empty = filled + rem;
        }
        return ans;
    }
}

File: 07-07-2024_Water_Bottles.py
class Solution:
    def numWaterBottles(self, numBottles: int, numExchange: int) -> int:
        empty = numBottles
        ans = numBottles
        while empty >= numExchange:
            filled = empty // numExchange
            rem = empty % numExchange
            ans += filled
            empty = filled + rem
        
        return ans
